## Завдання ПР 6  
У цій роботі ви спробуєте навчити класифікатор класифікувати позичальників банку на добросовісних (ті, що вчасно повернуть кредит) та недобросовісних (ті, що не повернуть кредит або повернуть невчасно) на основі набору характеристик позичальників банку.

In [2]:
import pandas as pd
df = pd.read_csv('credit_default.csv')
df.head()

person_age  person_income person_home_ownership  person_emp_length  \
0          22          59000                  RENT              123.0   
1          21           9600                   OWN                5.0   
2          25           9600              MORTGAGE                1.0   
3          23          65500                  RENT                4.0   
4          24          54400                  RENT                8.0   

  loan_intent loan_grade  loan_amnt  loan_int_rate  loan_status  \
0    PERSONAL          D      35000          16.02            1   
1   EDUCATION          B       1000          11.14            0   
2     MEDICAL          C       5500          12.87            1   
3     MEDICAL          C      35000          15.23            1   
4     MEDICAL          C      35000          14.27            1   

   loan_percent_income cb_person_default_on_file  cb_person_cred_hist_length  
0                 0.59                         Y                           3  
1                 0.10                         N                           2  
2                 0.57                         N                           3  
3                 0.53                         N                           2  
4                 0.55                         Y                           4

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32581 entries, 0 to 32580
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   person_age                  32581 non-null  int64  
 1   person_income               32581 non-null  int64  
 2   person_home_ownership       32581 non-null  object 
 3   person_emp_length           31686 non-null  float64
 4   loan_intent                 32581 non-null  object 
 5   loan_grade                  32581 non-null  object 
 6   loan_amnt                   32581 non-null  int64  
 7   loan_int_rate               29465 non-null  float64
 8   loan_status                 32581 non-null  int64  
 9   loan_percent_income         32581 non-null  float64
 10  cb_person_default_on_file   32581 non-null  object 
 11  cb_person_cred_hist_length  32581 non-null  int64  
dtypes: float64(3), int64(5), object(4)
memory usage: 3.0+ MB


### Опис датафрейму  
Цей набір даних містить інформацію про 32581 позичальників банку:  
**person_age** - вік позичальника  
**person_income** - річний дохід позичальника     
**person_home_ownership** - житло: орендує, виплачує іпотеку, власне житло     
**person_emp_length** - стаж роботи      
**loan_intent** - призначення кредиту: навчання, лікування і т.д.  
**loan_grade** - кредитний рейтинг       
**loan_amnt** - розмір кредиту     
**loanintrate** - відсоткова ставка по кредиту  
**loan_status - статус кредиту (0 сплачено, 1 - не сплачено) ЦІЛЬОВА ЗМІННА**     
**loan_percent_income** - розмір кредитного платежу у % до доходу   
**cb_person_default_on_file** - кредитна історія   
**cb_preson_cred_hist_length** - тривалість кредитної історії

### Завдання
1. Подивіться на датасет: .head (), .info (), .dtypes, .describe ()  
2. Для кращого розуміння даних порівняйте характеристики добросовісних та недобросовісних позичальників, **зробіть відповідні висновки**  
3. Виконайте попередню обробку даних  
4. Реалізуйте алгоритми логістичної регресії, найближчих сусідів (KNN) та опорних векторів (SVM) для класифікації позичальників банку (<span style="color:red"> перший алгоритм розглядала на лекції, другий, третій - на самостійне опрацювання</span>). Перевірте точність класифікаторів та обретіть найкращий із них для подальшої роботи. **Зробіть висновки: чи вдалося побудувати точний класифікатор; який із трьох розглянутих алгоритмів показав найкращі, а який найгірші результати?**  
5. Необхідно прийняти рішення щодо видачі кредиту  для нових позичальників, інформація про яких знаходиться в файлі 'give_refuse_a_loan.csv': спрогнозуйте значення цільової змінної та **зробіть висновки яким позичальникам варто видати кредит, а яким - відмовити.**